packages/echo-core/src/state/useGlobalState.ts

```diff
@@ -16,6 +15,6 @@ export function useGlobalState(): GlobalState;
 export function useGlobalState<R>(select: (state: GlobalState) => R): R;
 
 export function useGlobalState<R>(select?: (state: GlobalState) => R): GlobalState | R {
```

Comment:
I don't think we have unit tests for this, can you add this? Would be nice to test function and none function